functional/FunctionalTest.java

```diff
@@ -2421,16 +2421,9 @@ public static void testSetBucketPolicy(String objectPrefix, PolicyType policyTyp
         client.putObject(bucketName, objectName, new ByteArrayInputStream(data), data.length, customContentType);
       }
 
+      // Authenticated download is not supported in AWS S3.
       if ((policyType == PolicyType.READ_WRITE) || (policyType == PolicyType.WRITE_ONLY)) {
         writeObject(urlString, data);
-
-        InputStream is = client.getObject(bucketName, objectName);
```

Comment:
This check is doing whether anonymously uploaded object by `writeObject()` is downloadable by the authenticated user and its same content or not.

Could you tell me what exactly the problem is?

Reply:
It is not supported in AWS.

Reply:
How come an authenticated user not able to download objects whatever the way it is uploaded? I am sure the policy is wrongly set.  Can you confirm?

Reply:
This is very simple to test

```
$ mc mb s3/bala-123
Bucket created successfully `s3/bala-123`.
$ mc policy public s3/bala-123
Access permission for `s3/bala-123` is set to `public`

$ curl -X PUT -T /etc/issue https://bala-123.s3.amazonaws.com/public.txt
$ curl https://bala-123.s3.amazonaws.com/public.txt
Ubuntu 17.10 \n \l

$ mc cp s3/bala-123/public.txt /tmp
mc: <ERROR> Failed to copy `https://s3.amazonaws.com/bala-123/public.txt`. Insufficient permissions to access this file `https://s3.amazonaws.com/bala-123/public.txt`

$ mc cp --debug s3/bala-123/public.txt /tmp
...
...
mc: <DEBUG> HEAD /public.txt HTTP/1.1
Host: bala-123.s3.amazonaws.com
User-Agent: Minio (linux; amd64) minio-go/4.0.7 mc/DEVELOPMENT.GOGET
Authorization: AWS4-HMAC-SHA256 Credential=**REDACTED**/20180205/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20180205T081532Z

mc: <DEBUG> HTTP/1.1 403 Forbidden
Transfer-Encoding: chunked
Content-Type: application/xml
Date: Mon, 05 Feb 2018 08:15:31 GMT
Server: AmazonS3
X-Amz-Id-2: toz5YBQjx/oJrbXibScrFAN87EJBUTpCcLQRzmZOV+RcjCu/74VAaF0FlxkPZkHxuttxajTbHBA=
X-Amz-Request-Id: 129188C6471E3E70


mc: <DEBUG> Response Time:  190.267361ms

mc: <ERROR> Failed to copy `https://s3.amazonaws.com/bala-123/public.txt`. Insufficient permissions to access this file `https://s3.amazonaws.com/bala-123/public.txt`
 (3) cp-main.go:372 cmd.doCopySession(..) Tags: [https://s3.amazonaws.com/bala-123/public.txt]
 (2) common-methods.go:165 cmd.uploadSourceToTargetURL(..) Tags: [https://s3.amazonaws.com/bala-123/public.txt]
 (1) common-methods.go:93 cmd.getSourceStream(..) Tags: [s3, https://s3.amazonaws.com/bala-123/public.txt]
 (0) client-s3.go:926 cmd.(*s3Client).getObjectStat(..)
 Release-Tag:DEVELOPMENT.GOGET | Commit:DEVELOPMENT. | Host:prtsc | OS:linux | Arch:amd64 | Lang:go1.9.3 | Mem:2.6MB/9.5MB | Heap:2.6MB/5.7MB
```